src/bank.rs

```diff
@@ -759,30 +741,14 @@ impl Bank {
         Ok(accounts)
     }
 
-    /// Execute a transaction.
-    /// This method calls each instruction in the transaction over the set of loaded Accounts
-    /// The accounts are committed back to the bank only if every instruction succeeds
-    fn execute_transaction(
-        &self,
-        tx: &Transaction,
-        tx_accounts: &mut [Account],
-        tick_height: u64,
-    ) -> Result<()> {
-        for (instruction_index, instruction) in tx.instructions.iter().enumerate() {
-            let program_id = tx.program_id(instruction_index);
-            Self::with_subset(tx_accounts, &instruction.accounts, |program_accounts| {
-                let mut executable_accounts = self.load_executable_accounts(*program_id)?;
-                runtime::execute_instruction(
-                    tx,
-                    instruction_index,
-                    &mut executable_accounts,
-                    program_accounts,
-                    tick_height,
-                ).map_err(|err| BankError::ProgramError(instruction_index as u8, err))?;
-                Ok(())
-            })?;
-        }
-        Ok(())
+    /// For each program_id in the transaction, load its loaders.
+    fn load_loaders(&self, tx: &Transaction) -> Result<Vec<Vec<(Pubkey, Account)>>> {
+        tx.instructions
+            .iter()
+            .map(|ix| {
+                let program_id = tx.program_ids[ix.program_ids_index as usize];
+                self.load_executable_accounts(program_id)
```

Comment:
is this ok in the scheme of things?  this is outside self.lock_accounts(), right?

Reply:
Yes, it's okay because, at least currently, because the accounts are read-only and they are cloned before passing them to the program. Any changes to them will be discarded.